src/lib/utils/setNodeById.js

```diff
@@ -1,14 +1,13 @@
-function setNodeById(nodes, id, newNode) {
-  for (let i = 0; i < nodes.length; i++) {
-    const node = nodes[i];
-    if (node.id === id) {
-      nodes[i] = { id, ...newNode };
-    } else if (Array.isArray(node.nodes)) {
-      nodes[i].nodes = setNodeById(node.nodes, id, newNode);
-    }
+const setNodeById = (nodes, id, newNode) => nodes.map(node => {
+  let currentNode = node;
```

Comment:
We should make obj redestructring here or remove this variable because this is not different from the map's "node".